plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java

```diff
@@ -142,17 +142,22 @@ public static boolean isPrestoView(Map<String, String> tableParameters)
 
     public static boolean isHiveOrPrestoView(Table table)
     {
-        return isHiveOrPrestoView(table.getTableType());
+        return isHiveOrPrestoView(table.getTableType(), table.getParameters());
     }
 
-    public static boolean isHiveOrPrestoView(String tableType)
+    public static boolean isHiveOrPrestoView(String tableType, Map<String, String> tableParameters)
     {
-        return tableType.equals(VIRTUAL_VIEW.name());
+        return tableType.equals(VIRTUAL_VIEW.name()) && !isTrinoMaterializedView(tableParameters);
+    }
+
+    public static boolean isTrinoMaterializedView(Table table)
+    {
+        return isTrinoMaterializedView(table.getParameters());
     }
 
-    public static boolean isTrinoMaterializedView(String tableType, Map<String, String> tableParameters)
+    public static boolean isTrinoMaterializedView(Map<String, String> tableParameters)
     {
-        return isHiveOrPrestoView(tableType) && isPrestoView(tableParameters) && tableParameters.get(TABLE_COMMENT).equalsIgnoreCase(ICEBERG_MATERIALIZED_VIEW_COMMENT);
+        return isPrestoView(tableParameters) && tableParameters.get(TABLE_COMMENT).equalsIgnoreCase(ICEBERG_MATERIALIZED_VIEW_COMMENT);
```

Comment:
```
isHiveOrPrestoView(mv) // returns false
isPrestoView(mv) // returns true
```

however, per it's name, isHiveOrPrestoView should be superset of isPrestoView, right?

Reply:
Makes sense from a consistency standpoint. 

Removed `isHiveOrPrestoView` and the behavior can be replicated with `isViewOrMaterializedView(view) && !isTrinoMaterializedView(view)`. 

I think this makes sense from an API standpoint. I do not think a method like `isView` should exist because it is unclear if view refers to materialized or not.